updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -0,0 +1,357 @@
+# frozen_string_literal: true
+
+module Dependabot
+  class Updater
+    module Operations
+      class UpdateAllVersions
+        def self.applies_to?(job:)
+          return false if job.security_updates_only?
+          return false if job.updating_a_pull_request?
+          return false if job.dependencies&.any?
+
+          true
+        end
+
+        def initialize(service:, job:, dependency_snapshot:, error_handler:)
+          @service = service
+          @job = job
+          @dependency_snapshot = dependency_snapshot
+          @error_handler = error_handler
+          # TODO: Collect @created_pull_requests on the Job object?
+          @created_pull_requests = []
+        end
+
+        def perform
+          Dependabot.logger.info("Starting update job for #{job.source.repo}")
+          dependencies.each { |dep| check_and_create_pr_with_error_handling(dep) }
+        end
```

Comment:
I'd appreciate it if someone more deeply familiar with this repo, like @jakecoffman, could verify this workflow 🙏

Reply:
FWIW, this code is a copy-paste from the existing updater with some specific clauses removed - I think it's probably worth reviewing-by-commit: 

> The guts of this change are best looked reviewed via these specific commits:
> - 018fad2d6f1b7e0fa14a7b7b3336614ec4864421 This copy-pastes **all** of the methods from `Dependabot::Updater` that are required to pass the test suite with no modifications
> - 3f6ac3ad443b85dbb5d70db61e91b28292719494 In this commit, the `dependencies` method which parses and filters the job's dependency files has all logic related to Security updates and rebasing existing PRs removed
> - 3f6ac3ad443b85dbb5d70db61e91b28292719494 Removes all remaining Security-update specific code

Reply:
Looks good!

I haven't been watching the changes so forgive my lack of context in my questions, maybe you're already headed in this direction:

Would it make sense to make the Operation per-dependency? In `check_and_create_pr_with_error_handling` it calls `if pr_exists_for_latest_version?` and `if (existing_pr = ...`. In the job definition passed in, we can identify that a PR exists beforehand and so it should be straight forward to separate a `CreatePR` and `UpdatePR` Operations.

Reply:
> Would it make sense to make the Operation per-dependency?

This is a good question, the "fresh" Version Update is a funny special case as most of the other job types tend to focus on a single ( or handful ) of Dependencies.

I've been thinking more of Operations being a job-level abstraction for now, which might be a bit of a misnomer, but I definitely do think as we start to work in grouping we should start to split the create/update pr responsibilities out - I'm just not sure how to align that breakdown right now.